lua/checks.lua

```diff
@@ -0,0 +1,192 @@
+require("lua.compiler")
+require("lua.util")
+
+function CheckSystemFamily()
+	Check("Determining operating system family")
+	local path = os.getenv("PATH")
+
+	if not path then
+		Error()
+	end
+
+	-- Try find part of a absolute drive path like 'C:\Windows\System32'
+	dos = string.match(path, ":\\")
+
+	if dos then
+		local output = UniqueName()
+		os.execute("ver > " .. output)
+		handle = io.open(output)
+		if handle then
+		local output = handle:read("*a")
+			handle:close()
+			if string.find( string.lower(output), "windows" ) then
+				r = "Windows"
+			else
+				r = "Dos"
+			end
+		end
+		os.remove(output)
+	else -- Assume Posix compliant
+		r = "Unix"
+	end
+
+	Pass(r)
+	return r
+end
+
+function CheckEnvVar(var)
+	Check("Checking '" .. var .. "' enviroment variable")
+	v = os.getenv(var)
+
+	if not v then
+		Pass("None")
+	else
+		Pass("'" .. v .. "'")
+	end
+
+	return v
+end
+
+function CheckCompiler(family, makefile)
+	Compiler = {}
+	local tmpName = UniqueName()
+
+	if not CreateCTestFile(tmpName .. ".c") then Error() end
+
+	local cc = CheckEnvVar("CC")
+	if not cc then
+		Check("Guessing compiler")
+
+		if makefile == "cc" then Fail("None") -- Nothing was specified
+		elseif makefile == "watcom" then
+			Pass("Open Watcom")
+			CheckWatcomCompiler(tmpName)
+		else
+			Pass("GNU Compiler Collection or compatible")
+			if makefile == "clang" then
+				cc = "clang"
+			else
+				cc = "gcc"
+				CheckGccCompiler(cc, tmpName)
+			end
+		end
+	else
+		return CheckCustomCompiler(cc, tmpName)
+	end
+
+	os.remove(tmpName .. ".c")
+
+	return cc, cflags
+end
+
+function CheckRemoveFileCmd(family, filename)
+	local exec = family == "Unix" and "rm" or "DEL"
+	Check("Checking '" .. exec .. "' works")
+
+	local path = SanitizePath(family, filename .. "/2.txt")
+
+	file = io.open(path, "a")
+
+	if not file then Error() end
+
+	io.close(file)
+
+	local e = exec .. " " .. path
+	RunCommand(e)
+
+	file = io.open(path)
+	if file then
+		io.close(file)
+		Fail("No")
+	end
+
+	Pass("Yes")
+	return exec
+end
+
+function CheckRemoveDirCmd(family, filename)
+	local exec = family == "Unix" and "rm -R" or "DELTREE /Y"
```

Comment:
Not sure where/how this function is used. But no `DELTREE` on Windows. Ref: https://stackoverflow.com/questions/338895/what-ever-happened-to-deltree-and-whats-its-replacement

Reply:
This function is used to test which command works to delete a non-empty directory for the purposes of the `clean:` in the makefile it will later generate. Not testing for `RD /s /q` is an oversight. I'll fix that in the next bunch of commits

Reply:
OK, fine. I'll look forward to run the whole thing.

Reply:
285b4af has implemented a check for `RD /S /Q`. It is preferred over `DELTREE /Y` and tested for first